### DISCUSSION TITLE
question about multi cluster

### DISCUSSION THREAD

**Discussion by oravil** [coinfoundry/miningcore#1237] [General]
hey,

> i read on docs we can connect multi cluster with the main server.

**there is no any configurations on config examples to connect cluster to another.**

how is that?

**Comment by oravil** [comment]
"shareRelays"
i found it.

**Reply by MiningCryptoLive** [reply]
Care to explain for others?

**Reply by xmrstickers** [reply]
how does this look in the config?

**Reply by oliverw** [reply]
> how does this look in the config?

It's not that difficult. Master acts as share receiver and N worker nodes act as share relays. Its all pretty obvious if you study the configuration options a bit.

**Reply by magecrimson** [reply]
@oliverw agreed it is rather simple, though I guess a nice document may help. Though my guess anyone setting up a cluster should be pretty deep in MC at this point and shouldn't need one :L-)  my 2 cents

**Reply by oliverw** [reply]
> @oliverw agreed it is rather simple, though I guess a nice document may help. Though my guess anyone setting up a cluster should be pretty deep in MC at this point and shouldn't need one :L-) my 2 cents

And that my friend is exactly the reason why it is not documented.

**Reply by MiningCryptoLive** [reply]
My 2cents says that if you want new clients to use a product you supply the information needed to use the product. Now whether setting up a cluster is part of that product would depend on what you are leaving as open source and selling as a service.

**Reply by oliverw** [reply]
> My 2cents says that if you want new clients to use a product you supply the information needed to use the product. Now whether setting up a cluster is part of that product would depend on what you are leaving as open source and selling as a service.

It's not about a conflict of interest. Several MC users have been running clusters for a long time without purchasing it. It's not really a secret nor do I refuse to talk about. Information is just not front and center because I expect everyone moving into that direction to have a solid understanding of mining in general. I got too many inquiries for clustering solutions from people who didn't have the slightest clue about how mining pools work.

**Reply by xmrstickers** [reply]
Have been running on single-cluster production with my friends for a month with no downtime. API routed through HTTPS, all that good stuff. Miningcore is truly awesome! 
But, like many development questions, I asked to save myself some time, not because I think I am unable to do it on my own. 

I will dig through the config; thank you.

**Comment by MiningCryptoLive** [comment]
I hear ya. It's annoying to listen to the same questions over and over
again.

**Reply by oliverw** [reply]
> I hear ya. It's annoying to listen to the same questions over and over again.
> […](#)
> On Thu, Sep 15, 2022, 9:48 AM Oliver Weichhold ***@***.***> wrote: My 2cents says that if you want new clients to use a product you supply the information needed to use the product. Now whether setting up a cluster is part of that product would depend on what you are leaving as open source and selling as a service. It's not about a conflict of interest. Several MC user have been running clusters for a long time without purchasing it. It's not really a secret nor do I refuse to talk about. Information is just not front and center because I expect everyone moving into that direction to have a solid understanding of mining in general. I got too many inquiries for clustering solutions from people who didn't have the slightest clue about how mining pools work. — Reply to this email directly, view it on GitHub <[#1237 (reply in thread)](https://github.com/oliverw/miningcore/discussions/1237#discussioncomment-3654111)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAIAKRU5TJD5TNNVSYHEMRDV6MSJJANCNFSM5T7GAKOA> . You are receiving this because you commented.Message ID: ***@***.***>

Annoying is the wrong word. Some wishful pool operators are so blissfully clueless that you get strong urge to protect them from making a mistake.
